go/arrow/ipc/message.go

```diff
@@ -0,0 +1,121 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ipc // import "github.com/apache/arrow/go/arrow/ipc"
+
+import (
+	"fmt"
+	"sync/atomic"
+
+	"github.com/apache/arrow/go/arrow/internal/debug"
+	"github.com/apache/arrow/go/arrow/internal/flatbuf"
+	"github.com/apache/arrow/go/arrow/memory"
+)
+
+// MetadataVersion represents the Arrow metadata version.
+type MetadataVersion flatbuf.MetadataVersion
+
+const (
+	MetadataV1 = MetadataVersion(flatbuf.MetadataVersionV1) // version for Arrow-0.1.0
+	MetadataV2 = MetadataVersion(flatbuf.MetadataVersionV2) // version for Arrow-0.2.0
+	MetadataV3 = MetadataVersion(flatbuf.MetadataVersionV3) // version for Arrow-0.3.0 to 0.7.1
+	MetadataV4 = MetadataVersion(flatbuf.MetadataVersionV4) // version for >= Arrow-0.8.0
+)
+
+func (m MetadataVersion) String() string {
+	if v, ok := flatbuf.EnumNamesMetadataVersion[int16(m)]; ok {
+		return v
+	}
+	return fmt.Sprintf("MetadataVersion(%d)", int16(m))
+}
+
+// MessageType represents the type of Message in an Arrow format.
+type MessageType flatbuf.MessageHeader
+
+const (
+	MessageNone            = MessageType(flatbuf.MessageHeaderNONE)
+	MessageSchema          = MessageType(flatbuf.MessageHeaderSchema)
+	MessageDictionaryBatch = MessageType(flatbuf.MessageHeaderDictionaryBatch)
+	MessageRecordBatch     = MessageType(flatbuf.MessageHeaderRecordBatch)
+	MessageTensor          = MessageType(flatbuf.MessageHeaderTensor)
+	MessageSparseTensor    = MessageType(flatbuf.MessageHeaderSparseTensor)
+)
+
+func (m MessageType) String() string {
+	if v, ok := flatbuf.EnumNamesMessageHeader[byte(m)]; ok {
+		return v
+	}
+	return fmt.Sprintf("MessageType(%d)", int(m))
+}
+
+const (
+	// maxNestingDepth is an arbitrary value to catch user mistakes.
+	// For deeply nested schemas, it is expected the user will indicate
+	// explicitly the maximum allowed recursion depth.
+	maxNestingDepth = 64
+)
+
+// Message is an IPC message, including metadata and body.
+type Message struct {
+	ref  int64
```

Comment:
Why not name this field `refCount` to be coherent with the rest of the code base?

Reply:
done.